clients/src/test/java/org/apache/kafka/common/utils/UtilsTest.java

```diff
@@ -158,6 +161,39 @@ public void testCloseAll() {
         }
     }
 
+    @Test
+    public void testReadFully() throws IOException {
+        RandomAccessFile raf = new RandomAccessFile(TestUtils.tempFile(), "rw");
```

Comment:
I can use EasyMock to test throwing IOExceptions for EOF, how can I use it to test "readFully does not return until buffer if full"? A mocked ByteBuffer?